README.md
[![License](https://img.shields.io/badge/License-ASL%202.0-green.svg)](https://opensource.org/licenses/Apache-2.0)

# Websphere CCT 

**WebSphere Configuration Comparison Tool (WCCT)** is a light weight comparison utility for WebSphere Application Server traditional.  Use of this tool is a two step process.  The first step is gathering data from the configuration repository using the wsadmin script ConfigDump.py.  The second is generating HTML reports to display configuration differences.  Manually configured application servers and resources are likely to have configuration differences. Finding these differences by navigating to configuration panels is error prone and time consuming. Applications servers and resources should be periodically checked for configuration differences.  Note, please reference [WASConfigurationComparisonTool.pdf](WASConfigurationComparisonTool.pdf) for full documentation.  

WCCT provides a facility to check multiple server and resource configurations for:
- potential inconsistent behavior between application servers in a cluster.
- configuration creep.
- migration of configuration from version to version.
- file system differences.

## Current WCCT reports include:
- Application Servers
- Node Agents
- Global Security configuration (not Domain Security)
  - Trust Association (Global and Domain)
- SSL Configurations
- Data Sources
- J2C Resource Adapters
- JDBC Drivers
- Object Cache instances
- Servlet Cache instances
- Service Integration Bus instances
- Resource Environment Providers
- Resource Environment Entries
- JMS Queues, Topics Providers and Activation Specifications, JAAS Auth Data
- File system comparisons

## CCT usage scenarios:
- Isolating configuration creep: Periodic configuration snapshots can be gathered and archived. These can be used to see what configuration changes that have been made over time.
- Troubleshooting inconsistent cluster member configuration: Servers and node/server specific resource configuration anomalies can causes unexpected runtime behavior from different servers within a cluster.
- Assisting in migration to a new version of WebSphere: Cross version configuration reports can help ensure a complete migration of server and resource settings.

## Tar File Contents:
- [ConfigReport.py](ConfigReport.py): Script for creating reports for Traditional Profile WebSphere
- [ConfigDump.py](ConfigDump.py): wsadmin script for dumping configuration
- [ConfigUtils.py](ConfigUtils.py): Utilities script
- [WAuJ.py](WAuJ.py): Utilities script
- [WAuJ_utilities.py](WAuJ_utilities.py): Utilities script
- [ConfigReport.properties](ConfigReport.properties): Sample properties file for defining reports for Traditional Profile WebSphere
- [ConfigReportAttributes.properties](ConfigReportAttributes.properties): Defines attributes reported in Traditional Profile WebSphere reports.  It is recommended not to edit this.
- [CollectFileData.sh](CollectFileData.sh): Collects file data from the specified file system
- [ConfigReportFiles.py](ConfigReportFiles.py): Generates file system comparison reports
- [WASConfigurationComparisonTool.pdf](WASConfigurationComparisonTool.pdf): Tool documentation

## Contributing

This project uses the [Developer's Certificate of Origin 1.1 (DCO)](https://github.com/hyperledger/fabric/blob/master/docs/source/DCO1.1.txt) see [CONTRIBUTING.md](CONTRIBUTING.md) for more details.

## Change History:

See [CHANGELOG.md](CHANGELOG.md) for more details.

CHANGELOG.md
# Change History:

- February 17th, 2020
  - ConfigReport.py may now be run with no parameters as all necessary properties have been given default values.
- January 30th, 2020
  - Added various JMS reports.
  - Made ReportList and ConfigurationDumpFiles optional properties for simplified command line syntax.
  - Added capability to filter out top level object types from collection. 
- August 22nd 2019
  - More documentation updates. 
- August 21st 2019
  - Documentation updates. 
- August 19th 2019
  - Corrected report generation failure in SSL cipher reporting.
  - Added support for collecting file system data for AIX.
  - Added optional parameter on ConfigDump.py to set the cell suffix from the command line.
  - Improved syntax help on ConfigReportFiles.py
  - Added the ability to only collect data from specified scopes.
- June 6th 2019
  - Changed files that match in CRC and size but not date and time to be a conditional correct match.
  - Corrected issue with Trust Association report generation.
  - Disabled specific error messages in ConfigDump.py.
  - Added more documentation for file system reports.
  - Reordered documentation for better flow.
- April 3rd, 2019
  - Reports on Trust Association and Trust Association Interceptor configuration.
  - Light weight file system comparison using date, time and chksum of files.
- December 29th, 2018
  - Generate sample wsadmin scripts which will make configuration values consistent for resources identified in a given report. 
- November 14, 2018
  - Added application comparison reports.
    - Built in "Applications" report that compares configurations by deployed module name.
    - User defined reports that allow comparing any applications with each other.
  - Updated documentation.
    - Reordered presentation pages.
    - Added page for installation instructions
    - Added page detailing available resource reports
    - Added page detailing built in reports. 
- October 31, 2018
  - Corrected character truncation for generic JVM arguments as well as extraneous characters with leading and trailing arguments. 
- August 3, 2018
  - Resource Environment Provider reports.
  - Resource Environment Entry reports.
- August 2, 2018
  - Corrected issue in ConfigDump.py preventing it from running. 
  - Accounted for configuration anomalies with incomplete and orphaned variable substitution entries.
  - Corrected path separator issue on Linux for report generation.
- June 17th, 2018
  - Added ability to compare all SSL configurations, not just cell defaults.
  - Enhanced Resources report to compare SSL configurations by the same name automatically.
  - Added cipher report to compare which ciphers may be inconsistent between SSL configs.
  - Updated ConfigDump.py to explicitly gather configuration data for all SSL configurations. Configuration dumps done with prior versions of ConfigDump.py may not pick up all SSL configurations.
  - Enhanced Cell Summary report to list all SSL Configuration aliases and scopes.
  - Corrected a bug which broke individual JVM argument comparisons.
- June 2nd, 2018
  - New report properties file processing introduced.
  - Individual report properties may be set on the command line.
  - Reports may be abbreviated to attributes that have differences only.
  - A node agent report was added.
  - Thread Pool Manager and all thread pools are now reported.
- May 27th, 2018
  - Added a report key at the top of all report documenting values in match column and cell background colors.
  - Added capability to replace a string with another value or a regular expression for the purposes of matching.
  - Cell, node and server names are replaced with .* by default for matching purposes.
- January 19th, 2018
  - ConfigDump.py on Windows systems that do not have a Unix shell (sh) will fail. Added messages documenting a workaround.
  - Updated PDF presentation with ConfigReport.py prerequisites and syntax.
- January 17th, 2018
  - Initial publication


WAuJ.py
##########################################################################################
# 
# Copyright International Business Machines Corp. 2014, 2020.
# 
# See the NOTICE file distributed with this work for additional information
# regarding copyright ownership. Licensed under the Apache License,
# Version 2.0 (the "License"); you may not use this file except in compliance
# with the License. You may obtain a copy of the License at
# 
#   http://www.apache.org/licenses/LICENSE-2.0
# 
# Unless required by applicable law or agreed to in writing, software
# distributed under the License is distributed on an "AS IS" BASIS,
# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
# See the License for the specific language governing permissions and
# limitations under the License.
# 
#-------------------------------------------------------------------------------
# Name: WAuJ.py
# From: WebSphere Application Server Administration using Jython
#   By: Robert A. (Bob) Gibson [rag], Arthur Kevin McGrath, Noel J. Bergman
# ISBN: 0-13-700952-6
# Role: profile script to prepare wsadmin environment better
# History:
#   date   ver who what
# -------- --- --- -------------------------------------------------------------
# 11/05/20 0.3 rag Fix - Change how wasroot is determined since Jython 2.5.1
#                        initializes sys.path (very) differently
# 10/09/29 0.2 rag Add - Look for -profile *WAuJ.py command line option
#                        as suggested by Pete Neergaard <jpn@us.ibm.com>
#                        import WAuJ_utilities as WAuJ;   # if possible
# 09/10/29 0.1 rag Fix - Added ISBN detail
# 09/10/25 0.0 rag New - For the book
#-------------------------------------------------------------------------------

from __future__ import nested_scopes

#----------------------------------------------------------------------
# Name: msg()
# Role: Display the specified text message, but only if the global
#      "quiet" variable has a "false" value.
#----------------------------------------------------------------------
def msg( text ) :
  if not quiet :
    print text

#----------------------------------------------------------------------
# Add references for the WebSphere Admin objects to sys.modules
#----------------------------------------------------------------------
WSASobjects = { 'AdminApp'     : AdminApp    ,
                'AdminConfig'  : AdminConfig ,
                'AdminControl' : AdminControl,
                'Help'         : Help
              }
avail = 'Only some';                   # Default - Adminask isn't available

#----------------------------------------------------------------------
# If '-WAuJ' is specified as one of the command line arguments, then
# minimize the output generated by this profile script, and remove the
# value from sys.argv
#----------------------------------------------------------------------
try :
  sys.argv.remove( '-WAuJ' );
  quiet = 1;
except ValueError :
  quiet = 0;

msg( '-' * 79 );
msg( 'Actions performed by WAuJ.py:' );
msg( '> from __future__ import nested_scopes' );

#----------------------------------------------------------------------
# If AdminTask object isn't available, then we can't add a reference...
#----------------------------------------------------------------------
if 'AdminTask' in dir() :
  WSASobjects[ 'AdminTask' ] = AdminTask;
  avail = 'All';

#----------------------------------------------------------------------
# Shallow copy of the name/value pairs from the WSASobjects dictionary
# to the sys.modules dictionary.
#----------------------------------------------------------------------
sys.modules.update( WSASobjects );

#----------------------------------------------------------------------
# Display a message indicating how many of the Admin objects are available
#----------------------------------------------------------------------
msg( '> sys.modules.update( WSASobjects )  # %s admin objects added' % avail )

#----------------------------------------------------------------------
# Remove the unnecessary objects from the namespace
#----------------------------------------------------------------------
del WSASobjects, avail;

#----------------------------------------------------------------------
# If sys isn't in sys.modules, add it.
#----------------------------------------------------------------------
if 'sys' not in sys.modules :
  sys.modules[ 'sys' ] = sys;
  msg( "> sys.modules[ 'sys' ] = sys" );

#----------------------------------------------------------------------
# Determine the WebSphere Application Server installation root
# 1. Get the command line using java.lang.System.getenv( '...' )
#    C:\IBM\WebSphere\AppServer70\java\bin\java ...
# 2. Use split( ' ', 1 )[ 0 ] to copy this part of the string
# 3. Use split( os.sep )      to convert this to a list, ...
# 4. Use [ :-3 ]              to drop the last 3 fields, ...
# 5. Use os.sep.join()        to put it back together
#----------------------------------------------------------------------
import java;
import os;

cmdLine = java.lang.System.getenv( 'IBM_JAVA_COMMAND_LINE' );
wasroot = os.sep.join( cmdLine.split( ' ', 1 )[ 0 ].split( os.sep )[ :-3 ] )
msg( "> wasroot = r'%s'" % wasroot );

if 'scripts' in os.listdir( wasroot ) :
  sys.path.append( wasroot + os.sep + 'scripts' );
  msg( "> sys.path.append( r'%s' )" % sys.path[ -1 ] );

#----------------------------------------------------------------------
# If -profile .*WAuJ.py was on the commandline, add it's containg
# directory to sys.path (just in case that directory also contains
# WAuJ_utilties.py)
#----------------------------------------------------------------------
import re;

REpat   = re.compile( '.*-profile ([^ ]*WAuJ.py).*', re.IGNORECASE );
mo      = REpat.search( cmdLine );     # RegExp Match Object
if mo :
  WAuJ_dir = os.path.dirname( mo.group( 1 ) );
  if WAuJ_dir not in sys.path :
    sys.path.append( WAuJ_dir );
    msg( "> sys.path.append( r'%s' )" % WAuJ_dir );
  del WAuJ_dir;

#----------------------------------------------------------------------
# Try to add the WAuJ utilities module
#----------------------------------------------------------------------
try :
  import WAuJ_utilities as WAuJ;
  msg( '> import WAuJ_utilities as WAuJ' );
except :
  msg( '# Not found: WAuJ_utilities.py' );

#----------------------------------------------------------------------
# Remove the last of the temporary variables & modules (which were
# only needed for only a few things).
#----------------------------------------------------------------------
msg( '-' * 79 )

del os, msg, quiet, java, re, REpat, cmdLine, mo;


CollectFileData.sh
#!/bin/bash
##########################################################################################
# 
# Copyright International Business Machines Corp. 2014, 2020.
# 
# See the NOTICE file distributed with this work for additional information
# regarding copyright ownership. Licensed under the Apache License,
# Version 2.0 (the "License"); you may not use this file except in compliance
# with the License. You may obtain a copy of the License at
# 
#   http://www.apache.org/licenses/LICENSE-2.0
# 
# Unless required by applicable law or agreed to in writing, software
# distributed under the License is distributed on an "AS IS" BASIS,
# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
# See the License for the specific language governing permissions and
# limitations under the License.
# 
##########################################################################################
# Author: Dennis Riddlemoser
# Copyright IBM 2019 all rights reserved.
#
# Change history:
# Date        Who  Comments
# 2019/08/06  DWR  Added support for AIX istat
# 2019/08/06  DWR  Adressed div by zero error with less than ten files found
# 2019/08/07  DWR  Addressed files with embedded spaces.  
# 

if [ -z "$1" ]; then
	echo "Syntax: $0 [DIRECTORY] [OUTPUT_FILE]"
	echo "   DIRECTORY the directory to recursively collect file date/time/CRC information from"
	echo "   OUTPUT_FILE the file to write file data to"
	exit
	fi

directory=$1
outputFile=$2
if [ -z "$2" ]; then
	dateStr=`date +"%Y%m%d.%H%M%S"`
	outputFile="file-list.$dateStr.txt"
	fi

niceCmd="nice -n10"
statCmd=`which stat 2>nul`
if [ -z "$statCmd" ]; then
		statCmd=`which istat 2> nul`
		if [ -z "$statCmd" ]; then
			echo "AIX utilities found, using istat."
			fi
	else
		echo "GNU utilities found, using stat."
	fi
if [ -z "$statCmd" ]; then
	echo "Neither stat nor istat found.  File date/time comparisons cannot be done."
	fi

###############################################################################
# Get file last modified time/date and CRC
###############################################################################
echo "Getting last modification date and cksum for files.  This may take several minutes."
files=`$niceCmd find $directory -type f|grep -v /workarea/` 
numFiles=`echo $files|wc -w`
numFilesDiv10=$(($numFiles / 10))
if [ $numFilesDiv10 -eq 0 ]; then
	numFilesDiv10=1
	fi
echo "$numFiles files found."
count=0
while read -r file; do
	crc=`$niceCmd cksum "$file"|cut -f 1,2 -d" " 2>nul`
	if [ -z "$statCmd" ]; then
		date=""
	else
		date=`$niceCmd $statCmd "$file" 2>nul|grep  "odif"`
	fi
	echo "$date|$crc|$file" >> $outputFile
	count=$(($count+1))
	if [ $(($count % $numFilesDiv10)) -eq 0 ]; then
		echo "$(($count * 100 / $numFiles))% done."
		fi
	done <<< "$files"



ConfigReportFiles.py
##########################################################################################
# 
# Copyright International Business Machines Corp. 2014, 2020.
# 
# See the NOTICE file distributed with this work for additional information
# regarding copyright ownership. Licensed under the Apache License,
# Version 2.0 (the "License"); you may not use this file except in compliance
# with the License. You may obtain a copy of the License at
# 
#   http://www.apache.org/licenses/LICENSE-2.0
# 
# Unless required by applicable law or agreed to in writing, software
# distributed under the License is distributed on an "AS IS" BASIS,
# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
# See the License for the specific language governing permissions and
# limitations under the License.
# 
##########################################################################################
# Author: Dennis Riddlemoser
# This script generates reports based on tar files produced by ConfigDumpLiberty.sh
##########################################################################################
# Change history:
# Date        Ref  Who  Comments
# 2019/05/25  0030 DWR  Makeing date/time stapms a conditional match
# 2019/08/19  0036 DWR  Exit with syntax if less than two parameters on the command line
# 
# 
import ConfigUtils as cu
import sys
import time
import re
import os
import traceback
import glob
#Load libraries from where the path they are installed.
path = os.getenv('IBM_JAVA_COMMAND_LINE')
if path == None or len(path) == 0:
	path=sys.argv[0]
else:
	path=path[path.find("-f") + 2:].strip()
path=path.split(' ')[0]
if path.find(os.sep) > -1:
	path=path.split(' ')[0]
	path=path[:path.rfind(os.sep)+1]
else:
	path = ''

sys.path.append(path)
ScriptPath=path 

import ConfigUtils as cu

False=0
True=1


#########################################################################################################
#########################################################################################################
# Execution logic begins here 
#########################################################################################################
#########################################################################################################
if len(sys.argv) < 3:  #Ref0036
	print 'Syntax:'
	print '  python '+ sys.argv[0] + ' ReportFileName FileList1 FileList3 ...'
	print '    ReportFileName is the name of output report (fully qualified or relative)'
	print '    FileList parameters are lists of files generated by CollectFileData.sh' #Ref0036
#	print '    LibertyTarX is a tar file created by ConfigDumpLiberty.sh'
	sys.exit(0)

cu.printMsg('Begin run.',False)

fileIdx = 1
if sys.argv[0].find('.py') > 0:
	fileIdx = 2

ReportFileName = sys.argv[1]
cu.printMsg('Report file name: %s' % ReportFileName,False)

#Get information from input files
FileNames = sys.argv[fileIdx:]
for idx in range (0, len(FileNames)-1):
	FileNames[idx]=FileNames[idx].strip()
ServerDataList = cu.createServerFileDataList(FileNames) #Ref0030

#Run reports
cu.setReportOnlyMismatched("")
cu.reportInitialize('File System Report',ReportFileName)
cu.ReportFileSystem(ServerDataList)
cu.reportFinalize()

cu.printMsg('End run.',False)


ConfigDump.py
##########################################################################################
# 
# Copyright International Business Machines Corp. 2014, 2020.
# 
# See the NOTICE file distributed with this work for additional information
# regarding copyright ownership. Licensed under the Apache License,
# Version 2.0 (the "License"); you may not use this file except in compliance
# with the License. You may obtain a copy of the License at
# 
#   http://www.apache.org/licenses/LICENSE-2.0
# 
# Unless required by applicable law or agreed to in writing, software
# distributed under the License is distributed on an "AS IS" BASIS,
# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
# See the License for the specific language governing permissions and
# limitations under the License.
# 
##########################################################################################
# Author: Dennis Riddlemoser
# This script dumps out the majority of a server or cell configuration in a text format 
# for use with ConfigReport.py to compare configurations.
##########################################################################################
# Change history:
# Date        Ref  Who  Comments
# 2018/01/17  0001 DWR  Added logic to handle unsupported configuration types.
# 2018/01/19  0002 DWR  Added logic to handle systems where the environment is not present.
# 2018/06/16  0015 DWR  Collect all SSL config info.  
# 2019/01/30  0026 DWR  Using AdminConfig.types() to get configuration elements instead of hard coded list
# 2019/08/19  0037 DWR  Adding command line setting of CellSuffix
# 2019/08/19  0038 DWR  Limiting data gathered to specificed scope(s)
# 2020/01/09  0044 DWR  Added command line syntax to exclude collection specific objec types
#

import sys
import os

#Ref0002 Begin
def printException(msg,exceptionInfo):
	for line in exceptionInfo:
		msg=msg+'\n'+str(line)
	print 'ERROR: ' + msg

#Load libraries from where the path they are installed.
path = None
try:
	path = os.getenv('IBM_JAVA_COMMAND_LINE')
	path=path[path.find("-f") + 2:].strip()
	path=path.split(' ')[0]
	if path.find(os.sep) > -1:
		path=path.split(' ')[0]
		path=path[:path.rfind(os.sep)+1]
	else:
		path = ''
except:
	printException('Could not get command line from environment.',sys.exc_info())

sys.path.append(path)
try:
	import ConfigUtils as cu
except:
	printException('Failed to load support scripts.',sys.exc_info())
#Ref0002 End

#########################################################################################################
#########################################################################################################
# Gets configuration data from the repository
#########################################################################################################
#########################################################################################################
True=1
False=0

#Ref0002 Begin
try: 
	cu.printMsg('Begin run.',False)
except:
	print 'ERROR, cannot continue.  Failed to load support scripts.  Run wsadmin -f ConfigDump.py from the same directory it exists in.'
	sys.exit(1)

excludeTypeList = []  #Ref0044
regexList=[]  #Ref0038
if len(sys.argv) > 0:
	if " ".join(sys.argv).find("=") == -1: #Ref0038 
		cu.setRepositoryConfigFileName(sys.argv[0])
		#Ref0002 End
		if len(sys.argv) > 1:              #Ref0037
			cu.setCellSuffix(sys.argv[1])  #Ref0037
	#Ref0038 Begin
	else:
		for arg in sys.argv:
			if arg.find("="):
				parm = arg.split('=')[0]
				value = arg.split('=')[1]
				if parm == 'file':
					cu.setRepositoryConfigFileName(value)
				elif parm == 'suffix':
					cu.setCellSuffix(sys.argv[1])
				elif parm == 'excludeType':            #Ref0044
					excludeTypeList = value.split(',') #Ref0044
				else:
					regexList.append('.*%ss/%s[/|].*' % (parm,value))
			else:
				print('ERROR: Invalid parameter: %s' % arg)
	#Ref0038 End

# configElementList = [['Getting cluster configuration ','ServerCluster'],\
					 # ['Getting server configuration ','Server'],\
					 # ['Getting PMI configuration ','PMIModule'],\
					 # ['Getting Global Security configuration ','Security'],\
					 # ['Getting Security Domain configuration ','SecurityDomain'],\
					 # ['Getting SSL configuration ','SSLConfig'],  #Ref0015 \
					 # ['Getting Authorization Table ','AuthorizationTableExt'],\
					 # ['Getting Audit configuration ','Audit'],\
					 # ['Getting cell configuration ','Cell'],\
					 # ['Getting Core Group configuration ','CoreGroup'],\
					 # ['Getting Core Group Bridge configuration ','CoreGroupBridgeSettings'],\
					 # ['Getting cell variables ','VariableMap'],\
					 # ['Getting Virtual Host configuration ','VirtualHost'],\
					 # ['Getting cell Web Services Security configuration ','WSSecurity'],\
					 # ['Getting Object Pool Provider configuration ','ObjectPoolProvider'],\
					 # ['Getting Work Manager Provider configuration ','WorkManagerProvider'],\
					 # ['Getting Schedular Provider configuration ','SchedulerProvider'],\
					 # ['Getting JMS Provider configuration ','JMSProvider'],\
					 # ['Getting J2C Resource Adapter configuration ','J2CResourceAdapter'],\
					 # ['Getting Mail Provider configuration ','MailProvider'],\
					 # ['Getting URL Provider configuration ','URLProvider'],\
					 # ['Getting JDBC Provider configuration ','JDBCProvider'],\
					 # ['Getting Resource Environment Provider configuration ','ResourceEnvironmentProvider'],\
					 # ['Getting Data Source configuration ','DataSource'],\
					 # ['Getting Service Inigration Bus configuration ','SIBus'],\
					 # ['Getting JAX RPC handler configuration ','JAXRPCHandler'],\
					 # ['Getting Servlet Cache Instance configuration ','ServletCacheInstance'],\
					 # ['Getting Object Cache Instance configuration ','ObjectCacheInstance'],\
					 # ['Getting SIB configuration ','SIBus'],\
					 # ['Getting SIB WS Security Request Consumer Binding configuration ','SIBWSSecurityRequestConsumerBindingConfig'],\
					 # ['Getting SIB WS Security Request Generator Binding configuration ','SIBWSSecurityRequestGeneratorBindingConfig'],\
					 # ['Getting SIB WS Security Response Consumer Binding configuration ','SIBWSSecurityResponseConsumerBindingConfig'],\
					 # ['Getting SIB WS Security Response Generator Binding configuration ','SIBWSSecurityResponseGeneratorBindingConfig'],\
					 # ['Getting Application Deployment configuration ','ApplicationDeployment'],\
					 # ['Getting Resource Environment Providers','ResourceEnvironmentProvider'],\
					 # ['Getting Resource Environment Entries','ResourceEnvEntry'],\
					 # ]
configElementList = []
for type in AdminConfig.types('.*').splitlines():
	if excludeTypeList.count(type) == 0:    #Ref0044
		configElementList.append(type)
	else:                                                                #Ref0044
		cu.printMsg('Excluding configuration elements of type %s' % type,False) #Ref0044

for regex in regexList:                                              #Ref0038
	cu.printMsg('Filtering out configuration elements matching %s' % regex,False) #Ref0038	

#Ref0001 Moved logic into ConfigUtils.py
cu.getConfigElements(configElementList,AdminConfig,regexList) #Ref0038
#Ref0001 End
cu.writeRepositoryFile()


cu.printMsg('End run.',False)
cu.printMsg('Ignore the WASX7146I and WASX7309W messages.  No modifications to the repository were made so AdminConfig.save() was not invoked.',False)

print AdminConfig.queryChanges()
